enhancements/single-node-production-edge-cluster-profile.md

```diff
@@ -0,0 +1,469 @@
+having multiple sites provide coverage to overlapping geographic
+areas. Therefore, use of cluster-based high-availability features will
+be limited (for example, by running a single API service).
+
+Failures in edge deployments are frequently resolved by re-imaging or
+physically replacing the entire host. Combining this fact with the
+previous observation about the approach to providing highly-available
+services lets us draw the conclusion that in-place upgrades do not
+need to be supported by the first iteration of this cluster profile.
+
+The cluster profile will be sufficiently generic to cater to a variety
+of edge computing use cases. As such, OpenShift's usual cluster
+configuration mechanisms will be favored where there is a likelihood
+that there will be edge computing use case with differing requirements.
+For example, there is no expectation that this new profile will use
+the real-time kernel by default - this will continue to be a
+`MachineConfig` choice as per
+[enhancements/support-for-realtime-kernel](https://github.com/openshift/enhancements/blob/master/enhancements/support-for-realtime-kernel.md). *See open questions*
+
+### User Stories
+
+#### As a user, I can deploy OpenShift in a supported single-node configuration
+
+A user will be able to run the OpenShift installer to create a single-node
+deployment, with some limitations (see non-goals above). The user
+will not require special support exceptions to receive technical assistance
+for the features supported by the configuration.
+
+### Implementation Details/Notes/Constraints
+
+Some OpenShift components (such as Etcd and Ingress) require
+a minimum of 2 or 3 nodes. The `single-node-production-edge`
+cluster profile will configure these components as appropriate
+for a single node.
+
+When we are deploying a cluster with the `single-node-production-edge`
+cluster profile, the relevant operators should support a non-HA
+configuration that makes the correct adjustments to the deployment
+(e.g., `cluster-ingress-operator` should deploy a single router,
+`cluster-etcd-operator` should deploy the `etcd-member` [without
+waiting for 3 master
+nodes](https://github.com/openshift/cluster-etcd-operator/blob/98590e6ecfe282735c4eff01432ae40b29f81202/pkg/etcdenvvar/etcd_env.go#L72))
+
+In addition, some components are not relevant for this cluster profile
+(e.g. console, cluster-autoscaler, keepalived for ingressVIP and
+apiVIP) and shouldn't be deployed at all.
+
+#### cluster-etcd-operator
+
+By default, `cluster-etcd-operator` will not deploy the etcd cluster
+without minimum of 3 master nodes. This can be changed by enabling
+`useUnsupportedUnsafeNonHANonProductionUnstableEtcd`.
+
+```shell
+# allow etcd-operator to start the etcd cluster without minimum of 3 master nodes
+oc patch etcd cluster --type=merge -p="$(cat <<- EOF
+
+ spec:
+   unsupportedConfigOverrides:
+     useUnsupportedUnsafeNonHANonProductionUnstableEtcd: true
```

Comment:
What about finally adding a single toplevel API object that says "this is a single node cluster"?  If we're making this production it shouldn't involve `unsupportedConfigOverrides` for etcd.

Reply:
See #555 for a description of just such an API. :-)